File: src/01.lua
require 'src.common.common'

local huge = math.huge

function solve_1(numbers)
  local previous = huge
  local result = 0
  for n in all(numbers) do
    if n > previous then
      result = result + 1
    end
    previous = n
  end
  return result
end

function solve_2(numbers)
  local p1, p2, p3 = huge, huge, huge
  local result = 0
  for n in all(numbers) do
    if n + p1 + p2 > p1 + p2 + p3 then
      result = result + 1
    end
    p1, p2, p3 = n, p1, p2
  end
  return result
end

numbers = readlines('n')

print(solve_1(numbers))
print(solve_2(numbers))

File: src/01.rb
require './src/common/common.rb'

HUGE = 9999

def solve_1(numbers)
  result = 0
  previous = HUGE
  numbers.each do |n|
    if n > previous then
      result += 1
    end
    previous = n
  end
  result
end

def solve_2(numbers)
  result = 0
  p1 = p2 = p3 = HUGE
  numbers.each do |n|
    if n + p1 + p2 > p1 + p2 + p3 then
      result += 1
    end
    p3, p2, p1 = p2, p1, n
  end
  result
end

if __FILE__ == $0
  numbers = get_parsed_lines(lambda { |line| line.to_i })
  puts solve_1(numbers)
  puts solve_2(numbers)
end
